plugin/lighthouse/lighthouse.go

```diff
@@ -1,38 +1,16 @@
 package lighthouse
 
 import (
-	"context"
-
 	"github.com/coredns/coredns/plugin"
 	clog "github.com/coredns/coredns/plugin/pkg/log"
-	"github.com/coredns/coredns/plugin/pkg/nonwriter"
-	"github.com/coredns/coredns/request"
-
-	"github.com/miekg/dns"
 )
 
 // Define log to be a logger with the plugin name in it. This way we can just use log.Info and
 // friends to log.
 var log = clog.NewWithPlugin("lighthouse")
 
+type ServicesMap map[string]string
 type Lighthouse struct {
```

Comment:
I would add explicit compiler-time tests that make sure you have the correct interfaces implemented (e.g. error and Handler)

This is done by doing:

```
var _ error = &Lighthouse{}
var _ plugin.Handler = &Lighthouse{}
```

It also has the advantage to show whoever is reading your code that you intend and in fact implement those interfaces (and everyone knows that explicit is better than implicit ;))